### DISCUSSION TITLE
Question for creating own input forcing data

### DISCUSSION THREAD

**Discussion by yychengESM** [NCAR/hrldas#160] [Q&A]
I am preparing input forcing data to run a single point simulation at my site. May I ask:
(1) If I directly generate the forcing files as the final netcdf files, does it require to follow a certain format? For example, if my data starts from 1998-01-02 at a 30min interval, does it require writing forcing files for every 30min and naming them as: 199801020000.LDASIN_DOMAIN1, 199801020030.LDASIN_DOMAIN1, 199801020100.LDASIN_DOMAIN1, etc? Or is it possible to write the forcing into just one single netcdf file? If so, are there any places in the code that need modification?

(2) If I generate the forcing file by modifying the bondville.dat file and using create_point_data.exe, how should I deal with missing values in the bondville.dat file?

Thanks so much!

**Comment by cenlinhe** [comment]
For the default HRLDAS/Noah-MP, the forcing file name has to be the exact format. Also, it has to be each time point of forcing per file. You cannot write all forcing into just one single netcdf file. Yes, you can use 30min forcing.
If you want to modify the code to adopt different forcing file format, you could modify the code files under hrldas/hrldas/IO_code/. it is not that straighforward to read in one netcdf file with all forcing data.
For missing values in bondville.dat, you could set it to -9999, which is the default nan value used: https://github.com/NCAR/noahmp/blob/843a742c6019a613450f0002e7ec65f44e49b523/utility/Machine.F90
